dll/win32/shell32/CUserNotification.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Hermes Belusca-Maito
+ *
+ * Pass on icon notification messages to the systray implementation
+ * in the currently running shell.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef _USERNOTIFICATION_H_
+#define _USERNOTIFICATION_H_
+
+class CUserNotification :
+    public CComCoClass<CUserNotification, &CLSID_UserNotification>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IUserNotification
+//  public IUserNotification2   // On Vista+
+{
+private:
+    /* Private stuff... */
+
+public:
+    CUserNotification();
+    ~CUserNotification();
+
+    // IUserNotification
+    virtual HRESULT STDMETHODCALLTYPE SetBalloonInfo(
+        IN LPCWSTR pszTitle,
+        IN LPCWSTR pszText,
+        IN DWORD dwInfoFlags);
```

Comment:
In most COM classes we have the arguments on the same line as the function (not all of em).

Reply:
Yes, and it becomes unreadable/unmanageable. Hence I use the "new" technique of putting the arguments in new lines, as done in other parts of ROS.